examples/yaml_test.go

```diff
@@ -0,0 +1,241 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"path"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"os"
+	"os/exec"
+)
+
+const (
+	timeoutSeconds = 600
+	sleepBetween   = 10
+)
+
+// cmd will run the command c with args and if input is provided, that will be piped
+// into the process as input
+func cmd(t *testing.T, c string, args []string, input string) (string, error) {
+	t.Helper()
+	binary, err := exec.LookPath(c)
+	if err != nil {
+		t.Fatalf("couldn't find %s, plz install", c)
+	}
+
+	t.Logf("Executing %s %v", binary, args)
+
+	cmd := exec.Command(binary, args...)
+	cmd.Env = os.Environ()
+
+	if input != "" {
+		cmd.Stdin = strings.NewReader(input)
+	}
+
+	var stderr, stdout bytes.Buffer
+	cmd.Stderr = &stderr
+	cmd.Stdout = &stdout
+	if err := cmd.Run(); err != nil {
+		t.Fatalf("couldn't run command %s %v: %v, %s", c, args, err, stderr.String())
+		return "", err
+	}
+
+	return stdout.String(), nil
+}
+
+// resetState will erase all types that the yaml examples could have concievably created.
+func resetState(t *testing.T) {
+	t.Helper()
+
+	crdTypes := []string{
+		"conditions.tekton.dev",
+		"pipelineresources.tekton.dev",
+		"tasks.tekton.dev",
+		"clustertasks.tekton.dev",
+		"pipelines.tekton.dev",
+		"taskruns.tekton.dev",
+		"pipelineruns.tekton.dev",
+		"services",
+		"pods",
+		"configmaps",
+		"secrets",
+		"serviceaccounts",
+		"persistentvolumeclaims",
+	}
+	for _, c := range crdTypes {
+		cmd(t, "kubectl", []string{"delete", "--ignore-not-found=true", c, "--all"}, "")
+	}
+}
+
+// getYamls will look in the directory in examples indicated by version and run for yaml files
+func getYamls(t *testing.T, version, run string) []string {
+	t.Helper()
+	_, filename, _, _ := runtime.Caller(0)
+
+	// Don't read recursively; the only dir within these dirs is no-ci which doesn't
+	// want any tests run against it
+	files, err := ioutil.ReadDir(path.Join(path.Dir(filename), version, run))
+	if err != nil {
+		t.Fatalf("Couldn't read yaml files from %s/%s/%s: %v", path.Dir(filename), version, run, err)
+	}
+	yamls := []string{}
+	for _, f := range files {
+		if matches, _ := filepath.Match("*.yaml", f.Name()); matches {
+			yamls = append(yamls, f.Name())
+		}
+	}
+	return yamls
+}
+
+// replaceDockerRepo will look in the content f and replace the hard coded docker
+// repo with the on provided via the KO_DOCKER_REPO environment variable
+func replaceDockerRepo(t *testing.T, f string) string {
```

Comment:
Ugh, this is kinda gross. I think we could just drop any test that requires ko-building a package?

Reply:
we could - this is what the original yaml tests do tho, do you mind if i keep this in the context of this pull request and remove in another PR?